catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPreStoragePlugin.java

```diff
@@ -182,39 +204,79 @@ private void process(ContentItem contentItem, List<ContentItem> contentItems) {
-                            renderedImage.getWidth(),
-                            renderedImage.getHeight());
+                    ContentItem originalImageContentItem = createOriginalImage(contentItem.getId(),
+                            renderImageUsingOriginalDataModel(contentItem),
+                            metacard);
 
                     contentItems.add(originalImageContentItem);
                 }
+
             }
         } catch (IOException | ParseException | NitfFormatException | UnsupportedOperationException e) {
             LOGGER.debug(e.getMessage(), e);
         }
     }
 
     private BufferedImage renderImage(ContentItem contentItem)
-        throws IOException, ParseException, NitfFormatException {
+            throws IOException, ParseException, NitfFormatException {
+
+        return render(contentItem, input -> {
+            try {
+                return input.getRight()
+                        .render(input.getLeft());
+            } catch (IOException e) {
+                LOGGER.debug(e.getMessage(), e);
+            }
+            return null;
+        });
+
+    }
+
+    private BufferedImage renderImageUsingOriginalDataModel(ContentItem contentItem)
+            throws IOException, ParseException, NitfFormatException {
+
+        return render(contentItem, input -> {
+            try {
+                return input.getRight()
+                        .renderToClosestDataModel(input.getLeft());
+            } catch (IOException e) {
+                LOGGER.debug(e.getMessage(), e);
+            }
+            return null;
+        });
+
+    }
+
+    private BufferedImage render(ContentItem contentItem,
+            Function<Pair<ImageSegment, NitfRenderer>, BufferedImage> imageSegmentFunction)
+            throws IOException, ParseException, NitfFormatException {
 
         final ThreadLocal<BufferedImage> bufferedImage = new ThreadLocal<>();
 
-        if (contentItem != null && contentItem.getInputStream() != null) {
-            NitfRenderer renderer = new NitfRenderer();
-
-            new NitfParserInputFlow().inputStream(contentItem.getInputStream()).allData()
-                    .forEachImageSegment(segment -> {
-                        if (bufferedImage.get() == null) {
-                            try {
-                                bufferedImage.set(renderer.render(segment));
-                            } catch (IOException e) {
-                                LOGGER.debug(e.getMessage(), e);
-                            }
-                        }
-                    }).end();
+        if (contentItem != null) {
+            InputStream inputStream = contentItem.getInputStream();
+
+            if (inputStream != null) {
+                try {
+                    NitfRenderer renderer = new NitfRenderer();
+
+                    new NitfParserInputFlow().inputStream(inputStream)
+                            .allData()
+                            .forEachImageSegment(segment -> {
```

Comment:
You're reading each segment, but once the threadlocal has been set by segmentN you skip over all segments > segmentN. Can that be a very large list? If so, is there a way this could be written to short-circuit the all this no-op processing?